src/main/java/Duke.java

```diff
@@ -1,10 +1,202 @@
+        System.out.println(" Hello! I'm Duke" + System.lineSeparator()
+                + " What can I do for you?");
+        System.out.println(SEGMENT_LINE + System.lineSeparator());
+    }
+
+    public static void printBye() {
+        System.out.println(SEGMENT_LINE);
+        System.out.println(" Bye! Hope to see you again soon!");
+        System.out.print(SEGMENT_LINE + System.lineSeparator());
+    }
+
+    public static void runCommand(String input, Task[] tasks) {
+        String[] arrayOfInput = input.split(" ");
+        boolean isInputList = input.equals("list");
+        if (isInputList) {
+            listTasks(tasks);
+        } else if (isMark(arrayOfInput)) {
+            // if command is "mark <digit>"
+            markTask(tasks, arrayOfInput);
+        } else if (isUnmark(arrayOfInput)) {
+            // if command is "unmark <digit>"
+            unmarkTask(tasks, arrayOfInput);
+        } else {
+            addTask(input, tasks, arrayOfInput);
+        }
+    }
+
+    public static void listTasks(Task[] tasks) {
+        System.out.println(SEGMENT_LINE);
+        boolean isTaskCountZero = Task.totalTasks == 0;
+        if (isTaskCountZero) {
+            System.out.println(" The list is empty!");
+        } else {
+            System.out.println(" Here are the tasks in your lists:");
+            for (int i = 0; i < Task.totalTasks; i += 1) {
+                System.out.println(" " + (i + 1) + "." + tasks[i].getFullTaskDetail());
+            }
+        }
+        System.out.println(SEGMENT_LINE + System.lineSeparator());
+    }
+
+    public static boolean isMark(String[] arrayOfInput) {
+        boolean isTwoWordInput = arrayOfInput.length == 2;
+        boolean isFirstWordMark = arrayOfInput[0].equals("mark");
+        boolean isSecondWordNumber = isStringInteger(arrayOfInput[1]);
+        return (isTwoWordInput && isFirstWordMark && isSecondWordNumber);
+    }
+
+    public static boolean isUnmark(String[] arrayOfInput) {
+        // if input length (separated by " " is 2) && first word == "unmark" && second word only contains numbers
+        boolean isTwoWordInput = arrayOfInput.length == 2;
+        boolean isFirstWordUnmark = arrayOfInput[0].equals("unmark");
+        boolean isSecondWordNumber = isStringInteger(arrayOfInput[1]);
+        return (isTwoWordInput &&isFirstWordUnmark && isSecondWordNumber);
+    }
+
+    public static boolean isStringInteger(String input) {
+        // takes in a string and checks whether the string only contains digits
+        char[] inputInArray = input.toCharArray();
+        for (char c : inputInArray) {
+            if (!Character.isDigit(c)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static void markTask(Task[] tasks, String[] arrayOfInput) {
+        tasks[Integer.parseInt(arrayOfInput[1]) - 1].markAsDone();
+        printMarkOrUnmarkStatement(tasks, Integer.parseInt(arrayOfInput[1]) - 1);
```

Comment:
Readability: I think you should keep the function name shorter (maybe less than 20 characters)